orchestrator/crates/orchestrator/src/worker/utils/mod.rs

```diff
@@ -1,16 +1,76 @@
+use num_bigint::BigUint;
 use std::fmt::Write;
 use std::io::{BufRead, Cursor};
 use std::str::FromStr;
 use std::sync::Arc;
 
+use crate::core::config::Config;
+use crate::error::job::JobError;
+use crate::error::other::OtherError;
 use alloy::primitives::U256;
 use color_eyre::eyre::eyre;
-use num_bigint::BigUint;
 use starknet_os::io::output::StarknetOsOutput;
 
-use super::{JobError, OtherError};
-use crate::config::Config;
+pub mod da;
+pub mod fact_info;
+pub mod fact_node;
+pub mod fact_topology;
+pub mod helper;
+
+pub mod conversion;
+
+/// biguint_vec_to_u8_vec - Converts a vector of BigUint numbers to a vector of u8 bytes.
+///
+/// # Arguments
+/// * `nums` - A slice of BigUint numbers.
+///
+/// # Returns
+/// * `Vec<u8>` - A vector of u8 bytes representing the BigUint numbers.
+///
+pub fn biguint_vec_to_u8_vec(nums: &[BigUint]) -> Vec<u8> {
+    let mut result: Vec<u8> = Vec::new();
+
+    for num in nums {
+        result.extend_from_slice(biguint_to_32_bytes(num).as_slice());
+    }
+
+    result
```

Comment:
Please replace this with either a pre-allocated `Vec` or ideally an iterator. For example:
```suggestion
    nums.iter().map(biguint_to_32_bytes).map(array::as_slice).collect()
```

Reply:
updated the correct code for mapping